src/detect-engine-register.c

```diff
@@ -458,6 +439,17 @@ static void DetectFileHandlerRegister(void)
 void SigTableCleanup(void)
 {
     if (sigmatch_table != NULL) {
+        for (int i = 0; i < DETECT_TBLSIZE; i++) {
+            if ((sigmatch_table[i].flags & SIGMATCH_NEEDS_RUST_FREE) == 0) {
+                continue;
+            }
+            SCSigTableNamesElmt kw;
+            // remove const for mut to release
+            kw.name = (char *)sigmatch_table[i].name;
+            kw.desc = (char *)sigmatch_table[i].desc;
+            kw.url = (char *)sigmatch_table[i].url;
+            SCDetectSigMatchNamesFree(&kw);
+        }
```

Comment:
Would a `.Free` callback set in registration look/feel cleaner?

Reply:
Indeed, so that we can have C allocations as well
will do